packages/babel-parser/src/util/scope.js

```diff
@@ -38,13 +39,13 @@ export class Scope {
 // The functions in this module keep track of declared variables in the
 // current scope in order to detect duplicate variable names.
 export default class ScopeHandler<IScope: Scope = Scope> {
+  parser: Tokenizer;
   scopeStack: Array<IScope> = [];
-  declare raise: raiseFunction;
-  declare inModule: boolean;
+  inModule: boolean;
   undefinedExports: Map<string, Position> = new Map();
 
-  constructor(raise: raiseFunction, inModule: boolean) {
-    this.raise = raise;
+  constructor(parser: Tokenizer, inModule: boolean) {
+    this.parser = parser;
     this.inModule = inModule;
```

Comment:
I wasn't sure, and got scared by how `initializeScopes` in `UtilParse` seems to mutate `inModule` (it might end up behaving identically to just always referring to the parent parser's inModule, but wasn't as familiar with that part of the code). I spent a little time trying to understand all this when I briefly went down a rabbit hole of trying to get typescript module/namespace scopes [working better](https://github.com/tolmasky/babel/tree/parse-error-better-scopes), which I later abandoned because it would have made this PR much harder to understand I think, and because namespace scopes aren't super well documented (https://github.com/microsoft/TypeScript/issues/48047), but I do think there were some good ideas in there I'd like to return to. Anyways, that branch had better support for nesting scopes generally, to the point where I think we were close to being able to dump the "store everything in a temp variable, switch the parser, then restore everything in a finally block" strategy, and instead "push a module scope". In that sort of scenario, I think `inModule` makes more sense as *solely* belonging to the scope, vs. belonging to the parser.